examples/gcf-pubsub-vm-delete-event-handler/main.py

```diff
@@ -0,0 +1,660 @@
+        return self.MESSAGE.format(self.vm_uri, self.record['name'],
+                                   self.DETAIL.name)
+
+
+class NotARecord(RecordSkipped):
+    DETAIL = Detail.RR_NOT_A_RECORD
+
+
+class NameMismatch(RecordSkipped):
+    DETAIL = Detail.RR_NAME_MISMATCH
+
+
+class IpMismatch(RecordSkipped):
+    DETAIL = Detail.RR_IP_MISMATCH
+
+
+class RecordDeleted(StructuredLog):
+    MESSAGE = "{} matches DNS record {} deleted (RR_DELETED)"
+    SEVERITY = Severity.NOTICE
+    RESULT = Result.OK
+    LEVEL = logging.INFO
+    DETAIL = Detail.RR_DELETED
+
+    def __init__(self, vm_uri: str, event_id: str, dns_project: str,
+                 dns_managed_zone: str, record: dict, response: dict):
+        """The structure of record is a dns#resourceRecordSet
+
+        See: https://cloud.google.com/dns/docs/reference/v1/resourceRecordSets
+        """
+        super().__init__(vm_uri, event_id)
+        self.dns_project = dns_project
+        self.dns_managed_zone = dns_managed_zone
+        self.record = record
+        self.response = response
+
+    def message(self):
+        return self.MESSAGE.format(self.vm_uri, self.record['name'])
+
+    def info(self):
+        info = super().info()
+        info.update({
+            'dns_project': self.dns_project,
+            'dns_managed_zone': self.dns_managed_zone,
+            'dns_record': self.record,
+            'response': self.response,
+        })
+        return info
+
+
+class EventHandler():
+    """Handles a single event.
+
+    Intended to follow the lifecycle of a single trigger event.
+    """
+
+    def __init__(self, app, data, context=None):
+        self.config = self.load_configuration()
+        self.log = app.log
+        self.cloud_log = app.cloud_log
+        self.compute = app.compute
+        self.dns = app.dns
+        self.event_id = context.event_id if context else context
+        self.validate_data(data)
+        event = self.parse_data(data)
+        self.type = event['type']
+        self.event_subtype = event['event_subtype']
+        self.resource_type = event['resource_type']
+        self.project = event['project']
+        self.zone = event['zone']
+        self.vm_name = event['vm_name']
+        self.vm_uri = "projects/{}/zones/{}/instances/{}".format(
+            self.project, self.zone, self.vm_name
+        )
+        # https://cloud.google.com/functions/docs/env-var
+        self.function_project = os.getenv('GCP_PROJECT')
+        self.function_region = os.getenv('FUNCTION_REGION')
+        self.function_name = os.getenv('FUNCTION_NAME')
+        self.debug = True if os.getenv('DEBUG') else False
+
+    def load_configuration(self):
+        """Loads configuration from the environment
+
+        Returns:
+          Dictionary of config key/values.
+        """
+        dns_project = os.getenv('DNS_VM_GC_DNS_PROJECT')
+        if not dns_project:
+            raise(EnvironmentError(
+                'Env var DNS_VM_GC_DNS_PROJECT is required.'
```

Comment:
Use full words in Error messages.
```python
raise(EnvironmentError(
    'Environment Variable DNS_VM_GC_DNS_PROJECT is required.'
))
```

Also can we refactor this env var to not have `DNS` twice
s/DNS_VM_GC_DNS_PROJECT/DNS_VM_GC_PROJECT/

Reply:
Yes.  Related, the customer reported an issue.  Their managed zones are spread across multiple VPC networks, so I'd like to remove this configuration setting and instead make `DNS_VM_GC_DNS_ZONES` a list of fully qualified URI's.

Reply:
Fixed in 20d3bb1..ecde344